.github/workflows/browserstack.yml

```diff
@@ -2,6 +2,8 @@ name: BrowserStack
 
 on:
   push:
+    branches:
```

Comment:
We need to fix this to run on PRs if we are on the main repo

Reply:
If anyone has any ideas how to achieve this please let me know :)

Reply:
Pinging @julien-deramond in case you have any suggestions about the above. Basically, because the secrets are not available on forks, we need to allow PR runs only for people with push rights. Or only for the main repositories branches.

I never managed to solve this so any help is greatly appreciated :)

Reply:
you may want to put `v4-dev` , too

Reply:
> Pinging @julien-deramond in case you have any suggestions about the above.

Sent you a DM to have more details because I'm not sure to catch exactly the need.

But it seems that `if: github.repository == 'twbs/bootstrap'` already does the job.
When using a fork (people who doesn't have the push rights):
* if the target is for example julien-deramond/bootstrap:main, Browserstack job is skipped in my forked Bootstrap repo
* if the target is twbs/bootstrap:main, I don't see the Browserstack job run in my PRs on Bootstrap repo
Note: it would be possible to bypass that by modifying the workflow in a PR, but it is not common and still the secrets are not available.

Maybe something like that would fit the need:
```
on:
  push:
    branches:
      - main # runs only when code is pushed on the main branch
      - v4-dev # runs only when code is pushed on the v4-dev branch
  pull-request:
    branches:
      - main # run it when a PR has the main branch as a target (a PR with main-xmr-pa11y-ci as a target won't trigger it)
      - v4-dev # run it when a PR has the v4-dev branch as a direct target (a PR with main-xmr-pa11y-ci as a target won't trigger it)
  workflow_dispatch: # run it manually
  
jobs:
  browserstack:
    if: github.repository == 'twbs/bootstrap' # the jobs are skipped when try to run it from a fork
```

If we want to be sure about all of that we could create a fake repo with some _main_ and _v4-dev_ branches and do some tests.

Reply:
Thanks for the info! Indeed, I forgot I had the repo check.

So, I think the only thing left is to specify the `pull-request` branches, right?

Reply:
Yes I think. And target `v4-dev` as well if you find it relevant.

Reply:
I think `v4-dev` is enough if it's present in the v4-dev branch. We only need `v4-dev` for the CodeQL workflow's schedule event, I think.

Reply:
Maybe https://github.com/actions-cool/check-user-permission could help on this now ?